generic3g/OuterMetaComponent/get_child_name.F90

```diff
@@ -0,0 +1,33 @@
+#include "MAPL.h"
+
+submodule (mapl3g_OuterMetaComponent) get_child_name_smod
+
+   use mapl3g_GriddedComponentDriverMap
+   use mapl_ErrorHandling
+
+   implicit none
+
+contains
+
+   module function get_child_name(this, index, rc) result(name)
+      class(OuterMetaComponent), target, intent(in) :: this
+      integer, intent(in) :: index
+      integer, optional, intent(out) :: rc
+      character(len=:), allocatable :: name
+
+      type(GriddedComponentDriverMapIterator) :: iter
+      integer :: i
+
+      _ASSERT(index > 0, "index should be >= 1")
+      _ASSERT(index <= this%get_num_children(), "index should be <= num_children")
+
+      iter = this%children%ftn_begin()
+      do i = 1, index
```

Comment:
@tclune I thought there was an overloaded `next(index)`, but that did not seem to work. So looping till `index`

Reply:
There is such an interface - I see it in the source code.   My apologies - looking more carefully, I now see that the interface is called `advance`.

```
   iter = this%children%ftn_begin()
   call iter%advance(n)
```
I'll do a `suggestion`